consensus/parlia/parlia.go

```diff
@@ -868,6 +868,9 @@ func (p *Parlia) assembleVoteAttestation(chain consensus.ChainHeaderReader, head
 	voteAddrSet := make(map[types.BLSPublicKey]struct{}, len(votes))
 	signatures := make([][]byte, 0, len(votes))
 	for _, vote := range votes {
+		if _, ok := voteAddrSet[vote.VoteAddress]; ok {
+			return fmt.Errorf("assemble VoteAttestation failed, double sign by %s", common.Bytes2Hex(vote.VoteAddress[:]))
```

Comment:
Would it be possible that a validator could have 2 votes with same `vote.Data`?
since we have already checked the `vote.Data`

Reply:
this check both work for 'same votes' and 'different votes' with the same targetNumber from one VoteAddress

Reply:
@brilliant-lx u r right, no need check here